src/Microsoft.AspNetCore.Mvc.Core/ServiceFilterAttribute.cs

```diff
@@ -58,14 +57,11 @@ public IFilterMetadata CreateInstance(IServiceProvider serviceProvider)
                 throw new ArgumentNullException(nameof(serviceProvider));
             }
 
-            var service = serviceProvider.GetRequiredService(ServiceType);
-
-            var filter = service as IFilterMetadata;
-            if (filter == null)
+            var filter = (IFilterMetadata)serviceProvider.GetRequiredService(ServiceType);
+            if (filter is IFilterFactory filterFactory)
             {
-                throw new InvalidOperationException(Resources.FormatFilterFactoryAttribute_TypeMustImplementIFilter(
```

Comment:
Why we remove the error checking?

Reply:
Making the code in the two types consistent. The cast should throw just as well and it doesn't seem like the exception message would have been super useful:

> The type provided to ServiceFilterAttribute must implement IFilterMetadata.

Doesn't say what type. IMO, doing this check in the ctor is a bunch more useful so the user could see it fail at the call site rather than somewhere during filter execution.

Reply:
alirhgt